### ISSUE TITLE
Loss printing 

### ISSUE THREAD

**Issue by Murtazaabidi1** [danielsyahputra/yolo_distillation#6]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

@danielsyahputra 
I hope you are doing well. I have been working on your repository and have a question regarding the implementation of CWD and MGD in the trainer.py file. Specifically, I noticed that the following code snippet is provided for training the method:

from ultralytics import YOLO

teacher_model = YOLO("teacher_llvip_best.pt")
student_model = YOLO("yolo11n-obb.pt")

student_model.train(
    data="/home/murtaza/Desktop/Work/YOLOv11/datasets/std_LLVIP_RGB/data.yaml",
    teacher=teacher_model.model,
    distillation_loss="cwd",
    epochs=100,
    batch=16,
    workers=0,
    exist_ok=True,
    imgsz=640,
)

However, I couldn’t find where the distillation loss (CWD and MGD) is logged or printed during training. Is it being tracked or displayed anywhere, such as during inference or in the training logs? If not, could you please guide me on how to monitor or print the distillation loss during training? Because When I run the code through the above code the cmd run the yolo training as normal one

**Comment by danielsyahputra** [comment]
Hi @Murtazaabidi1 

You can find the part of how I calculate the distillation loss in `ultralytics/engine/trainer.py`

![image](https://github.com/user-attachments/assets/28b19e9b-2e67-4fb4-894a-107fdaa6b745)

> Because When I run the code through the above code the cmd run the yolo training as normal one

Yes, its because the distillation loss is added to the original loss with respect to its own weight. Please note that in Knowledge Distillation problem, we will calculate two type of loss. The first one is the loss of the student model towards the labels. The second one is the loss of model towards the teacher model.

So, these two losses will be summed based on their own weight. Thats why you don't see the distillation loss printed in the training output.

**Comment by Murtazaabidi1** [comment]
@danielsyahputra  Thank you for the clarification regarding the computation of the distillation loss. As you mentioned, the distillation loss is combined with the original loss based on their respective weights. During training, the logs typically display metrics such as box_loss, cls_loss, and dfl_loss. My question is:

Is the distillation loss incorporated into the total loss for each batch during training, or is it calculated and added after processing an entire batch? Additionally, could you confirm if this process influences the real-time training metrics displayed during each epoch?

**Comment by danielsyahputra** [comment]
@Murtazaabidi1 the distillation loss is calculated after processing each batch same as how we calculate the original loss. Using this way, we try to applied the distillation for each mini batch not batch optimization. By using this approach, we can save a lot of memory from processing the entire batch. Though, mini batch processing will introduce some noise instead of batch processing.

**Comment by Murtazaabidi1** [comment]
@danielsyahputra Thanks for you kind response it is very helpful to all people.

**Comment by danielsyahputra** [comment]
@Murtazaabidi1 sure, I'm happy if this work can help you.
